### DISCUSSION TITLE
CKB Mainnet Node (on external SSD)

### DISCUSSION THREAD

**Discussion by joshyates1980** [nervosnetwork/ckb#4169] [Q&A]
My goal is to install and run CKB full node 24hrs/7days on my [Surface 2](https://en.wikipedia.org/wiki/Surface_2) with an external USB3 500gb SSD drive. I have executed full nodes (without Neuron) many times over the past 2 years, but only by the recommendations from [nervos documentation](https://docs.nervos.org/docs/basics/guides/mainnet).

When I attempt to execute: `D:\ckb\ckb_v0.110.2_x86_64-pc-windows-msvc` with any command such as `ckb init` or `ckb run` a runtime error occurs from ckb.exe. The error dialog box states: VCRUNTIME140.dll was not found. VCRUNTIME140_1.dll was not found. MSVCP140.dll was not found.

Do I need to find those dll files and move them over to the D: drive?
Can I execute a ckb full node on an external ssd?
Is it possible to execute a ckb full node on other specified drives other than C: drive?

In addition, I'm unclear if I should post troubleshoot/guide questions on github, so if the primary Nervos developers would rather me post on [Nervos talk forum](https://talk.nervos.org/t/ckb-mainnet-node-on-external-ssd/7408?u=joshyates1980) moving forward, I'll honor your request. Thank you.

**Comment by code-monad** [comment]
Hello joshyates1980! Have you ever tried install the [Visual C++ Redistributable](https://www.microsoft.com/en-US/download/details.aspx?id=48145) on your device? 

If so, you may find these DLLs on your `C:\Windows\System32` (through File Explorer's "find bar"). If you haven't, try install the redist package first, then open a new terminal window, the commands should be fine now.

**Reply by joshyates1980** [reply]
@code-monad This was my fix:

I installed [Visual C++ Redistributable packages for Visual Studio 2015, 2017, 2019, and 2022 2](https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170#visual-studio-2015-2017-2019-and-2022) and now the CKB full node runs successfully on my external hard drive.
